File: src/main/java/sales_by_match/SalesByMatch.java
package sales_by_match;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashSet;

public class SalesByMatch {
    public void salesByMatch(){
        HashSet<Integer> socksList = new HashSet<Integer>(Arrays.asList(10,20,20,10,10,30,50,10,20));
        ArrayList<Integer> list = new ArrayList<Integer>(socksList);
        int pair_count = 0;
        for(int i = 0; i<=socksList.size(); i++){
            int element = list.get(i);
            if(socksList.contains(element)){
                socksList.remove(element);
                pair_count ++;
            }else{
                socksList.add(element);
            }
        }
        System.out.println(pair_count);
    }
}

File: src/main/kotlin/sales_by_match/salesByMatch.kt
package sales_by_match

import java.util.*


fun salesByMatch(){
    val socksList = HashSet(listOf(10, 20, 20, 10, 10, 30, 50, 10, 20))
    val list = ArrayList(socksList)
    var pairCount = 0

    for (i in socksList.indices){
        val element = list[i]
        if (socksList.contains(element)){
            socksList.remove(element)
            pairCount++
        }else{
            socksList.add(element)
        }
    }
    println(pairCount)
}